### PR TITLE
Allow generators/blueprints to force skipGeneratedFlag

### PR DESCRIPTION
**Pull request by sendilkumarn**
Moving the `skip-generated-flag` check during the runtime. Adding a sub-generator for `fu` but this blocks me from generating the application. 

Also it looks like we are adding & checking this flag in the base generator, I think this should be in the server, entity-server, spring-{service, controller} generator only. 

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by mshima — CHANGES_REQUESTED** [review]
With this change skipGeneratedFlag will not be persisted.
Will require the user to pass --skip-generated-flag every time.

I don't get why this is failing.
Can you give an example?

**Comment by mshima** [comment]
> Also it looks like we are adding & checking this flag in the base generator, I think this should be in the server, entity-server, spring-{service, controller} generator only.

This would create more problems.
- Conflicter is shared between every generator.
- Only the conflicter of the first generator is used.
- generators only register to it's own conflicter.

So:
- if app the generator doesn't register this generator-flag, it will not will be used at all when executing `jhipster`.
- if only app register the transformer, other generators will not use this transformer at all.

See https://github.com/jhipster/generator-jhipster/pull/12711.
Currently `jhipster entity Foo` generates java code without been prettified.

**Comment by sendilkumarn** [comment]
@mshima so for example, I have a new generator say `foo`.  Now when I generate the application, I have the `@GeneratedByJHipster` annotation. But I do not need this annotation.  

Setting the flag inside the `foo` generator do not work because before invoking `foo` generator it is invoked in `base` generator.

**Comment by sendilkumarn** [comment]
> Currently jhipster entity Foo generates java code without been prettified.

I do not want adding annotation but I do need prettier transform. Why `skip-generated-flag` influence the prettifying?

**Comment by mshima** [comment]
> > Currently jhipster entity Foo generates java code without been prettified.
> 
> I do not want adding annotation but I do need prettier transform. Why `skip-generated-flag` influence the prettifying?

Nothing, just using as example since the same principles applies.
Transformers are global, they are applied when committing the file from mem-fs to the disk.

**Comment by mshima** [comment]
> @mshima so for example, I have a new generator say `foo`. Now when I generate the application, I have the `@GeneratedByJHipster` annotation. But I do not need this annotation.

I don't understand the use case when you are saying that you don't want this.
If a `.java` file is written using the generator, it will add the flag.
Why do you want to prevent some `.java` file from having the flag?

**Comment by mshima** [comment]
I saw 2 problem with current implementation:
- adds the flags do maven downloader 
- adds the flag to package-info.java.

We should prevent from adding the flag to:
- java files outside packageFolder.
- package-info.java.

**Comment by sendilkumarn** [comment]
The whole point of fu is moving away from the `annotations` now we are adding them here. I am trying to circumvent that.

File: generators/generator-transforms.js [hunk 1]
```diff
@@ -51,7 +51,12 @@ const prettierTransform = function (defaultOptions) {
 
 const generatedAnnotationTransform = generator => {
     return through.obj(function (file, encoding, callback) {
-        if (path.extname(file.path) === '.java' && file.state !== 'deleted' && !file.path.endsWith('GeneratedByJHipster.java')) {
+        if (
+            !generator.options.skipGeneratedFlag &&
```
Comment (mshima) [inline generators/generator-transforms.js:55]:
```suggestion
            !generator.configOptions.skipGeneratedFlag &&
            !file.path.endsWith('package-info.java') &&
            !file.path.endsWith('MavenWrapperDownloader.java') &&
```

**Review by mshima — CHANGES_REQUESTED** [review]
You should revert the changes to generator-base.js.

And change from `options` to `configOptions`.
`options` is generator specific and will be applied for the entry generator only.
`configOptions` is shared between every generator so you can apply from every generator.

**Review by mshima — APPROVED** [review]

**Comment by sendilkumarn** [comment]
Thanks @mshima ! There are a lot of changes!

**Pull request merged**